### PR TITLE
Add memory object and interface

### PR DESCRIPTION
**Pull request by kesselb**

### CODE REVIEW

File: lib/OperatingSystems/IOperatingSystem.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	 * @return Memory
+	 */
+	public function getMemory(): Memory;
+
+	/**
+	 * Get name of the processor
+	 *
+	 * @return string
+	 */
+	public function getCPUName(): string;
```
Comment (ChristophWurst) [inline lib/OperatingSystems/IOperatingSystem.php:43]:
```suggestion
	public function getCpuName(): string;
```

debatable, but I think that reads better

File: lib/Resources/Memory.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\ServerInfo\Resources;
+
+class Memory {
+	public $memTotal = -1;
```
Comment (ChristophWurst) [inline lib/Resources/Memory.php:29]: how about setters and getters and `null` for undefined values?

File: lib/OperatingSystems/IOperatingSystem.php (hunk 1 above)
Comment (kesselb) [reply lib/OperatingSystems/IOperatingSystem.php:43]: Fine by me.

File: lib/Resources/Memory.php (hunk 2 above)
Comment (kesselb) [reply lib/Resources/Memory.php]: done

**Pull request merged**